scripts/bin/update-packages

```diff
@@ -148,22 +151,130 @@ _fetch_and_cache_tags() {
 	echo "INFO: Fetching and caching tags"
 
 	# First invocation of termux_repology_api_get_latest_version fetches and caches repology metadata.
-	quiet termux_repology_api_get_latest_version ' '
+	#quiet termux_repology_api_get_latest_version ' '
 
-    local __PACKAGES=()
+	local __PACKAGES=()
+	local __GITHUB_PACKAGES=()
 	for repo_dir in $(jq --raw-output 'del(.pkg_format) | keys | .[]' "${TERMUX_SCRIPTDIR}/repo.json"); do
 		for pkg_dir in "${repo_dir}"/*; do
-			! quiet _should_update "${pkg_dir}" && continue # Skip if not needed.
+			quiet _should_update "${pkg_dir}" || continue # Skip if not needed.
 			grep -q '^termux_pkg_auto_update' "${pkg_dir}/build.sh" && continue # Skip if package has custom auto-update
-			__PACKAGES+=("${pkg_dir}")
+			if grep '^TERMUX_PKG_SRCURL=' "${pkg_dir}/build.sh" | grep -q 'github.com'; then
+				__GITHUB_PACKAGES+=("${pkg_dir}")
+			else
+				__PACKAGES+=("${pkg_dir}")
+			fi
 		done
 	done
 
+	local GITHUB_GRAPHQL_QUERIES=() COUNTER=0
+	declare -A __GITHUB_GRAPHQL_PACKAGES=()
+	# Ignore non-constant sources
+	# shellcheck disable=SC1091
+	for pkg_dir in "${__GITHUB_PACKAGES[@]}"; do
+		local PKG_SRCURL TAG_TYPE OWNER REPO
+		read -r PKG_SRCURL TAG_TYPE < <(
+			set +u
+			source "${pkg_dir}/build.sh"
+			echo "${TERMUX_PKG_SRCURL} ${TERMUX_PKG_UPDATE_TAG_TYPE}"
+		)
+
+		read -r OWNER REPO < <(sed -E 's|^(git\+)?https?://github.com/([^/]+)/([^/]+).*|\2 \3|' <<< "$PKG_SRCURL")
+
+		if [[ -z "${TAG_TYPE}" ]]; then # If not set, then decide on the basis of url.
+			if [[ "${PKG_SRCURL:0:4}" == "git+" ]]; then
+				# Get newest tag.
+				TAG_TYPE="newest-tag"
+			else
+				# Get the latest release tag.
+				TAG_TYPE="latest-release-tag"
+			fi
+		fi
+
+		# This holds the source repositories, and query type for each repo we need to query.
+		# Newlines need to be escaped for the query parser
```

Comment:
We can strip newlines and hard tabs in the final HTTP request composing with something like `tr -d '\n\t'` and drop this restriction in here.

Reply:
As below, yes we can.
I still think it's important to have in mind.